### PR TITLE
CMake: Reorganize configure_bundled() definition to bundled subdirectory

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

File: bundled/setup_bundled.cmake [hunk 1]
```diff
@@ -42,6 +75,21 @@ option(DEAL_II_FORCE_BUNDLED_KOKKOS
 
 set(KOKKOS_FOLDER "${CMAKE_SOURCE_DIR}/bundled/kokkos-3.7.00")
 
+macro(feature_kokkos_configure_bundled)
+  set(KOKKOS_VERSION "3.7.0")
+  set(Kokkos_DEVICES "<bundled>")
```
Comment (masterleinad) [inline bundled/setup_bundled.cmake:80]:
```suggestion
  set(Kokkos_DEVICES "Serial")
```

File: bundled/setup_bundled.cmake [hunk 2]
```diff
@@ -42,6 +75,21 @@ option(DEAL_II_FORCE_BUNDLED_KOKKOS
...
+  set(Kokkos_ARCH "<bundled>")
```
Comment (masterleinad) [inline bundled/setup_bundled.cmake:81]:
```suggestion
  set(Kokkos_ARCH "none")
```
or
```suggestion
  set(Kokkos_ARCH " ")
```

**Comment by tamiko** [comment]
@masterleinad Updated, merci!

**Review by masterleinad — APPROVED** [review]

**Comment by tamiko** [comment]
In reference to #12374

**Pull request merged**